src/frontends/src/query_frontend.hpp

```diff
@@ -156,16 +156,34 @@ class RequestHandler {
                                         "New application detected: {}", key);
             auto app_info =
                 clipper::redis::get_application_by_key(redis_connection_, key);
-            std::vector<std::string> candidate_model_names =
-                clipper::redis::str_to_model_names(
-                    app_info["candidate_model_names"]);
             InputType input_type =
                 clipper::parse_input_type(app_info["input_type"]);
             std::string policy = app_info["policy"];
             std::string default_output = app_info["default_output"];
             int latency_slo_micros = std::stoi(app_info["latency_slo_micros"]);
-            add_application(name, candidate_model_names, input_type, policy,
-                            default_output, latency_slo_micros);
+            add_application(name, input_type, policy, default_output,
+                            latency_slo_micros);
+          }
+        });
+
+    clipper::redis::subscribe_to_model_link_changes(
+        redis_subscriber_,
+        [this](const std::string& key, const std::string& event_type) {
+          std::string app_name = key;
+          clipper::log_info_formatted(
+              LOGGING_TAG_QUERY_FRONTEND,
+              "APP LINKS EVENT DETECTED. App name: {}, event_type: {}",
+              app_name, event_type);
+          if (event_type == "sadd") {
+            clipper::log_info_formatted(LOGGING_TAG_QUERY_FRONTEND,
+                                        "New model link detected for app: {}",
+                                        app_name);
+
+            boost::optional<std::vector<std::string>> linked_model_names =
+                clipper::redis::get_linked_models(redis_connection_, app_name);
+            if (linked_model_names) {
```

Comment:
This method may not be necessary (see my comment below about whether or not it's a good idea to store link information locally)